shared/actions/chat.js

```diff
@@ -342,23 +384,66 @@ function * _setupNewChatHandler (): SagaGenerator<any, any> {
 const followingSelector = (state: TypedState) => state.config.following
 
 function * _loadInbox (): SagaGenerator<any, any> {
-  const inbox: GetInboxAndUnboxLocalRes = ((yield call(localGetInboxAndUnboxLocalRpcPromise, {param: {
-    query: {
-      status: Object.keys(CommonConversationStatus).filter(k => !['ignored', 'blocked'].includes(k)).map(k => CommonConversationStatus[k]),
+  const channelConfig = singleFixedChannelConfig([
+    'chat.1.chatUi.chatInboxUnverified',
+    'chat.1.chatUi.chatInboxConversation',
+    'chat.1.chatUi.chatInboxFailed',
+    'finished',
+  ])
+
+  const loadInboxChanMap: ChannelMap<any> = localGetInboxNonblockLocalRpcChannelMap(channelConfig, {
+    param: {
+      query: {
+        status: Object.keys(CommonConversationStatus).filter(k => !['ignored', 'blocked'].includes(k)).map(k => CommonConversationStatus[k]),
+        computeActiveList: true,
+        tlfVisibility: CommonTLFVisibility.private,
+        topicType: CommonTopicType.chat,
+        unreadOnly: false,
+        readOnly: false,
+      },
+      identifyBehavior: TlfKeysTLFIdentifyBehavior.chatGui,
     },
-    identifyBehavior: TlfKeysTLFIdentifyBehavior.chatGui,
-  }})): any)
-  const author = yield select(usernameSelector)
+  })
 
-  const following = yield select(followingSelector)
+  const chatInboxUnverified = yield takeFromChannelMap(loadInboxChanMap, 'chat.1.chatUi.chatInboxUnverified')
 
+  if (!chatInboxUnverified) {
+    throw new Error("Can't load inbox")
+  }
+
+  const inbox: GetInboxLocalRes = chatInboxUnverified.params.inbox
+  const author = yield select(usernameSelector)
+  const following = yield select(followingSelector)
   const conversations: List<InboxState> = _inboxToConversations(inbox, author, following || {})
   yield put({type: Constants.loadedInbox, payload: {inbox: conversations}})
+  chatInboxUnverified.response.result()
+
+  // +1 for the finish call
+  const total = inbox.conversationsUnverified && inbox.conversationsUnverified.length + 1 || 0
+  for (let i = 0; i < total; ++i) {
```

Comment:
surprising there isn't an ES6 `for (let i of range(0, total)) { .. }` kind of thing!  Looks like the closest would be:
```js
for (let i of Array(total).keys()) { .. }
```